synapsebridgehelpers/export_tables.py

```diff
@@ -0,0 +1,342 @@
+    Parameters
+    ----------
+    source_cols : list of synapseclient.Column objects
+    target_cols : list of synapseclient.Column objects
+    source_table : pandas.DataFrame
+    target_table : pandas.DataFrame
+    rename_threshold : float
+        When comparing column values between `source_table` and `target_table`,
+        if a proportion greater than the `rename_threshold` is shared between
+        the columns, these column are considered to be the same column, even
+        if they have different names. This results in the columns being labled
+        as "renamed". Values are compared both on actual value and index, up to
+        the length of the shorter column.
+
+    Returns
+    -------
+    A dictionary containing keys:
+
+    added
+    removed
+    modified
+    renamed
+
+    The value of renamed is a key mapping the column name in the target table
+    to the new column name in the source table. The rest of the values are sets.
+    """
+    comparison = {}
+    source_cols_dic = {c["name"]: c for c in source_cols}
+    target_cols_dic = {c["name"]: c for c in target_cols}
+    added_cols = {c["name"] for c in source_cols if c not in target_cols}
+    removed_cols = {c["name"] for c in target_cols if c not in source_cols}
+    modified_cols = {c for c in added_cols if c in removed_cols}
+    for c in modified_cols:
+        added_cols.discard(c)
+        removed_cols.discard(c)
+    renamed_cols = {}
+    if (source_table is not None and target_table is not None and
+        len(added_cols) and len(removed_cols)):
+        for source_col in added_cols:
+            for target_col in removed_cols:
+                if (source_cols_dic[source_col]["columnType"] ==
+                    target_cols_dic[target_col]["columnType"]):
+                    if source_cols_dic[source_col]["columnType"] == "FILEHANDLEID":
+                        raise Exception("A column containing file handles "
+                                        "was potentially renamed.")
+                    overlap = [i == j for i, j in
+                               zip(source_table[source_col],
+                                   target_table[target_col])]
+                    if sum(overlap) / len(overlap) >= rename_threshold:
+                        renamed_cols[target_col] = source_col
+        for target_col, source_col in renamed_cols.items():
+            removed_cols.discard(target_col)
+            added_cols.discard(source_col)
+    comparison["renamed"] = renamed_cols
+    comparison["modified"] = modified_cols
+    comparison["removed"] = removed_cols
+    comparison["added"] = added_cols
+    return comparison
+
+
+def synchronize_schemas(syn, schema_comparison, source, target,
+                        source_cols=None, target_cols=None):
+    """Update (on Synapse) a target Schema to match a source Schema.
+
+    Parameters
+    ----------
+    syn : synapseclient.Synapse
+    schema_comparison : dict
+    source_table : pandas.DataFrame
```

Comment:
Descrive what schema_comparison is.  It looks like the description is in the wrong place.